APM > .NET Tracer (#11200)
DataDog/documentation: content/en/tracing/setup_overview/compatibility_requirements/dotnet-framework.md

      tag: 'GitHub'
      text: 'Examples of Custom Instrumentation'
---
- The .NET Tracer supports all .NET-based languages (for example, C#, F#, Visual Basic).

- The .NET Tracer library for Datadog is open-source. For more information, see the [tracer Github repository][1].

## Supported .NET Framework runtimes
The .NET Tracer supports instrumentation on .NET Framework 4.5 and above (using CLR v4.0). It also supports [.NET Core][2].

| Version  | Microsoft End of Life |
| -------- | --------------------- |
| 4.8      |                       |
| 4.7.2    |                       |
| 4.7      |                       |
| 4.6.2    |                       |
| 4.6.1    | 04/26/2022            |
| 4.6      | 04/26/2022            |
| 4.5.2    | 04/26/2022            |
| 4.5.1    | 01/12/2016            | 
| 4.5      | 01/12/2016            |

>  Additional information on .NET Framework support policy can be found at [Microsoft .NET Framework Lifecyle Policy][3]. 

Reviewer: ```suggestion
 Additional information on .NET Framework support policy can be found within [Microsoft's .NET Framework Lifecycle Policy][3]. 
```

      tag: 'GitHub'
      text: 'Examples of Custom Instrumentation'
---
<div class="alert alert-warning"> 

>   <strong>Notes:</strong><br><ul><li>Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution within your application environment.</li><li> If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.</li></ul>

Reviewer 2: ```suggestion
  <strong>Notes:</strong><br><ul><li>Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution in your application environment.</li><li> If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.</li></ul>
```

</div>

## Compatibility
- The .NET Tracer supports all .NET-based languages (for example, C#, F#, Visual Basic).

- The .NET Tracer library for Datadog is open-source. For more information, see the [tracer Github repository][1].

### Supported .NET Framework runtimes
The .NET Tracer supports instrumentation on .NET Framework 4.5 and above (using CLR v4.0). It also supports [.NET Core][2].

| Version  | Vendor End of Life |
| -------- | ------------------ |
| 4.8      |                    |
| 4.7.2    |                    |
| 4.7      |                    |
| 4.6.2    |                    |
| 4.6.1    | 04/26/2022         |
| 4.6      | 04/26/2022         |
| 4.5.2    | 04/26/2022         |
| 4.5.1    | 01/12/2016         | 
| 4.5      | 01/12/2016         |

 Additional information on .NET Framework support policy can be found at [Microsoft .NET Framework Lifecyle Policy][3]. 

> ### Supported Process Architecture

Reviewer 2: ```suggestion
### Supported Processor Architectures
```

> ### Supported Processor Architectures

Author: replaced

| MSMQ                            | built-in                                                                                  | `Msmq`               |
| MySql                           | `MySql.Data` 6.7.0+                                                                       | `AdoNet`             |
| Oracle                          | `Oracle.ManagedDataAccess` 4.122.0+                                                       | `AdoNet`             |
| PostgreSQL                      | `Npgsql` 4.0+                                                                             | `AdoNet`             |
| RabbitMQ                        | `RabbitMQ.Client` 3.6.9+                                                                  | `RabbitMQ`           |
| Redis (ServiceStack client)     | `ServiceStack.Redis` 4.0.48+                                                              | `ServiceStackRedis`  |
| Redis (StackExchange client)    | `StackExchange.Redis` 1.0.187+                                                            | `StackExchangeRedis` |

> | SQL Server                      | `System.Data` 4.0.0+, `System.Data.SqlClient` 4.0.0 +, `Microsoft.Data.SqlClient` 1.0.0+  | `AdoNet`             |

Reviewer 2: We want to force line breaks here. See [this comment](https://github.com/DataDog/documentation/pull/11200/files#r676861334).

```suggestion
| SQL Server                      | `System.Data` 4.0.0+</br>`System.Data.SqlClient` 4.0.0+</br>`Microsoft.Data.SqlClient` 1.0.0+  | `AdoNet`             |
```

      tag: 'GitHub'
      text: 'Examples of Custom Instrumentation'
---
<div class="alert alert-warning"> 
  <strong>Notes:</strong><br><ul><li>Datadog automatic instrumentation relies on the .NET CLR Profiling API. This API allows only one subscriber (for example, APM). To ensure maximum visibility, run only one APM solution within your application environment.</li><li> If you are using both automatic and custom instrumentation, it is important to keep the package versions (for example, MSI and NuGet) in sync.</li></ul>
</div>

## Compatibility
Datadog .NET application performance monitoring supports a robust set of features to provide critical information about .NET applications & services: 

- The .NET Tracer supports all .NET-based languages (for example, C#, F#, Visual Basic).

- Monitor [runtime metrics][1] to help troubleshoot bugs and detect resource inefficiencies in your application.

- Ability to correlate [traces & application logs][2].

- The .NET Tracer library for Datadog is open-source. For more information, see the [tracer Github repository][3].

### .NET runtimes
The .NET Tracer supports instrumentation for [.NET Core][4] & NET Framework.

> | Supported .NET Framwork Versions | Vendor End of Life |

Reviewer 3: On this table you went from oldest to newest, but the .NET Core you went to newest to oldest. We should standardize on one order. I personally prefer the other approach so at the top of the table you see the latest versions of the runtime.

Author: Good catch thanks! Newer up top - works for me. I'll make the adjustments later.

Author: Updated the .NET Framework from newer on top to older at bottom. Changes are [here](https://github.com/DataDog/documentation/pull/11200/commits/d4260a023194b2e0791277b5c9242b0903fff21e).

Marking as resolved.
